handler/handler.go

```diff
@@ -0,0 +1,154 @@
+package handler
+
+import (
+	"encoding/json"
+	"log"
+	"main/repo"
+	"net/http"
+	"time"
+
+	"github.com/go-chi/chi/v5"
+	"github.com/go-chi/chi/v5/middleware"
+)
+
+type ServiceInterface interface {
+	CreateRecord(request repo.TransformRequest) *repo.Record
+	GetRecord(id string) (*repo.Record, error)
+	GetRecords() ([]repo.Record, error)
+	UpdateRecord(id string, request repo.TransformRequest) *repo.Record
+	DeleteRecord(id string) error
+}
+
+type Handler struct {
+	service ServiceInterface
+}
+
+type DBLayer interface {
+	NewRecord(r *repo.Record) error
+	GetRecord(id string) (repo.Record, error)
+	GetRecords() ([]repo.Record, error)
+	UpdateRecord(r *repo.Record) error
+	DeleteRecord(id string) error
+}
+
+func NewHandler(service ServiceInterface) *Handler {
+	return &Handler{
+		service: service,
+	}
+}
+
+func (h *Handler) RunServer() {
+	router := chi.NewRouter()
+	router.Use(middleware.Logger)
+	router.Post("/records", h.NewRecord)
+	router.Get("/records", h.GetAllRecords)
+	router.Get("/records/{id}", h.GetRecord)
+	router.Delete("/records/{id}", h.DeleteRecord)
+	router.Put("/records/{id}", h.UpdateRecord)
+
+	//nolint:exhaustivestruct, exhaustruct
+	server := &http.Server{
+		Addr:              ":8080",
+		Handler:           router,
+		ReadHeaderTimeout: 3 * time.Second,
+	}
+
+	err := server.ListenAndServe()
+	if err != nil {
+		log.Fatal("server listenig error")
+	}
+}
+
+func ResponseWithJSON(w http.ResponseWriter, result *repo.Record, results *[]repo.Record, statusCode int) {
+	w.Header().Add("Content-Type", "application/json")
+	w.WriteHeader(statusCode)
+
+	var err error
+
+	if result != nil {
+		err = json.NewEncoder(w).Encode(result)
+	} else {
+		err = json.NewEncoder(w).Encode(results)
+	}
+
+	if err != nil {
+		http.Error(w, err.Error(), http.StatusInternalServerError)
+		return
+	}
+}
+
+func (h *Handler) NewRecord(w http.ResponseWriter, r *http.Request) {
+	request := new(repo.TransformRequest)
+
+	err := json.NewDecoder(r.Body).Decode(&request)
+	if err != nil {
+		http.Error(w, err.Error(), http.StatusInternalServerError)
+		return
+	}
+
+	err = request.Validate()
+	if err != nil {
+		http.Error(w, err.Error(), http.StatusBadRequest)
```

Comment:
`http.StatusBadRequest` - >`http.StatusUnprocessableEntity`

Bad Request stands for "server cannot understand what request is"
while Unprocessable Entity - "server read the request but it's invalid and cannot be processed"

https://developer.mozilla.org/en-US/docs/Web/HTTP/Status/400
https://developer.mozilla.org/en-US/docs/Web/HTTP/Status/422